string.c

```diff
@@ -3630,6 +3630,55 @@ rb_str_concat_multi(int argc, VALUE *argv, VALUE str)
     return str;
 }
 
+static VALUE
+rb_str_append_bytes(VALUE str, VALUE str2)
+{
+    Check_Type(str2, T_STRING);
+
+    long str2_len = RSTRING_LEN(str2);
+
+    if (RB_UNLIKELY(str2_len == 0)) {
+        return str;
+    }
```

Comment:
Nitpick, but is this test needed for correctness? If not, might as well remove it, because why would people do this?

Reply:
It's not required no. As for why people do this, well, generally when you combine strings together you are not checking if any of them is empty:

```ruby
user.city + user.address_line_1 + user.address_line_2 + etc
```

But yeah, we could decide it's the rare case and remove that one check.